app/src/main/kotlin/com/mgorski/discountasciiwarehouse/asciiitemlist/AsciiItemListViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.mgorski.discountasciiwarehouse.asciiitemlist
+
+import android.databinding.ObservableArrayList
+import android.databinding.ObservableBoolean
+import com.mgorski.discountasciiwarehouse.R
+import com.mgorski.discountasciiwarehouse.model.AsciiItem
+import com.mgorski.discountasciiwarehouse.network.AsciiItemsProvider
+import com.mgorski.discountasciiwarehouse.view.messages.MessagesManager
+import com.mgorski.discountasciiwarehouse.view.recyclerview.pagination.PaginationRecyclerViewOnScrollListener
+import rx.android.schedulers.AndroidSchedulers
+
+class AsciiItemListViewModel(private val asciiItemsProvider: AsciiItemsProvider, private val messagesManager: MessagesManager) {
+
+    private var query = ""
+    private var onlyInStock = false
+    private val items: MutableList<AsciiItem> = arrayListOf()
+
+    val filteredItems = ObservableArrayList<AsciiItem>()
+    var isLoading = ObservableBoolean(false)
+
+    init {
+        loadItems()
+    }
+
+    val onLoadMoreCommand = { listener: PaginationRecyclerViewOnScrollListener -> loadItems(listener) }
+
+    fun onQueryChangedCommand(query: String) {
+        this.query = query
+        items.clear()
+        filteredItems.clear()
+        loadItems()
+    }
+
+    fun onFilterChangedCommand(onlyInStock: Boolean) {
+        this.onlyInStock = onlyInStock
+        updateFilteredList()
+    }
+
+    private fun loadItems(listener: PaginationRecyclerViewOnScrollListener? = null): Unit {
+        isLoading.set(true)
+
+        asciiItemsProvider.getAsciiItems(skip = items.count(), tagsQuery = query)

```

Comment:
It seems like you can pass the `onlyInStock` param here instead of filtering the resulting list, is there a reason for not doing that?